<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>993</width>
    <height>572</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3" columnstretch="6,4">
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>日期时间</string>
     </property>
     <layout class="QGridLayout" name="gridLayout" columnstretch="1,4,4,1">
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>时间</string>
        </property>
        <property name="buddy">
         <cstring>timeEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QTimeEdit" name="timeEdit">
        <property name="displayFormat">
         <string>hh:mm:ss</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QPushButton" name="btnGetDateTime">
        <property name="text">
         <string>获取当前日期时间</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLineEdit" name="lineEditDateTime"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>日期时间</string>
        </property>
        <property name="buddy">
         <cstring>dateTimeEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDateTimeEdit" name="dateTimeEdit">
        <property name="displayFormat">
         <string>yyyy-MM-dd hh:mm:ss</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLineEdit" name="lineEditTime"/>
      </item>
      <item row="1" column="3">
       <widget class="QPushButton" name="btnSetTime">
        <property name="text">
         <string>设置时间</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLineEdit" name="lineEditDate"/>
      </item>
      <item row="2" column="1">
       <widget class="QDateEdit" name="dateEdit">
        <property name="displayFormat">
         <string>yyyy年M月d日</string>
        </property>
        <property name="calendarPopup">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>日期</string>
        </property>
        <property name="buddy">
         <cstring>dateEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QPushButton" name="btnSetDate">
        <property name="text">
         <string>设置日期</string>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QPushButton" name="btnSetDateTime">
        <property name="text">
         <string>设置日期时间</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_4">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>字符串显示</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buddy">
         <cstring>timeEdit</cstring>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1" rowspan="2">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>日历选择</string>
     </property>
     <widget class="QCalendarWidget" name="calendarWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>296</width>
        <height>236</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>81</width>
        <height>28</height>
       </rect>
      </property>
      <property name="text">
       <string>选择的日期</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEditSel">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>30</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
     </widget>
    </widget>
   </item>
   <item row="1" column="0" rowspan="2">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>定时器</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,0,0,0" columnstretch="1,0,1,0,8">
      <item row="0" column="0" colspan="2">
       <widget class="QPushButton" name="btnStart">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>开始</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2" colspan="2">
       <widget class="QPushButton" name="btnStop">
        <property name="text">
         <string>停止</string>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QLabel" name="labelOverTime">
        <property name="text">
         <string>流逝时间</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>定时周期</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="2">
       <widget class="QSpinBox" name="spinBoxMs">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="suffix">
         <string> ms</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="value">
         <number>1000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="3" colspan="2">
       <widget class="QPushButton" name="btnSetMs">
        <property name="text">
         <string>设置周期</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="5">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLCDNumber" name="lcdHour"/>
        </item>
        <item>
         <widget class="QLCDNumber" name="lcdMin"/>
        </item>
        <item>
         <widget class="QLCDNumber" name="lcdSec"/>
        </item>
       </layout>
      </item>
      <item row="3" column="0" colspan="5">
       <widget class="QProgressBar" name="progressBar">
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QPushButton" name="btnExit">
     <property name="text">
      <string>退出</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>btnExit</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>732</x>
     <y>442</y>
    </hint>
    <hint type="destinationlabel">
     <x>718</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
